apps/files/src/store/mutations.js

```diff
@@ -244,5 +244,13 @@ export default {
       }
     })
     state.files = []
+  },
+
+  SET_CURRENT_SIDEBAR_TAB(state, tab) {
+    state.currentSidebarTab = tab
+  },
+
+  SET_CURRENT_COLLABORATORS_TAB_PANEL(state, panel) {
```

Comment:
IMO this is too specific. What about an options object as an optional second argument for the current sidebar tab? Then we would be able to pass in context specific data without changing the store everytime we add something to the sidebar.

Could be called like this: `SET_CURRENT_SIDEBAR_TAB(state, { tab, options: {'collaborators-tab-sub-panel': 'newCollaborator'} }`

Reply:
I still introduced mutation though which sets only options and not the whole tab object. This is because in collaborators we often change only the panel without changing the tab and it would feel strange to always have to define the collaborators' tab all over again.